add query_by_tokens option in Neural Sparse Search (#7040)
opensearch-project/documentation-website: _search-plugins/neural-sparse-search.md

When selecting a model, choose one of the following options:

- Use a sparse encoding model at both ingestion time and search time (high performance, relatively high latency).

> - Use a sparse encoding model at ingestion time and a tokenizer at search time (relatively low performance, low latency). The tokenizers doesn't conduct model inference, but for consistent experience we still deploy and invoke them using ml-commons model APIs.

Reviewer: ```suggestion
- Use a sparse encoding model at ingestion time and a tokenizer at search time for relatively low performance and low latency. The tokenism doesn't conduct model inference, therefore for a more consistent experience, deploy and invoke tokenizer using the ML commons Model API.
```

> - Use a sparse encoding model at ingestion time and a tokenizer at search time for relatively low performance and low latency. The tokenism doesn't conduct model inference, therefore for a more consistent experience, deploy and invoke tokenizer using the ML commons Model API.

Reviewer 2: ```suggestion
- Use a sparse encoding model at ingestion time and a tokenizer at search time for relatively low performance and low latency. The tokenism doesn't conduct model inference, so you can deploy and invoke a tokenizer using the ML Commons Model API for a more consistent experience.
```

Before the document is ingested into the index, the ingest pipeline runs the `sparse_encoding` processor on the document, generating vector embeddings for the `passage_text` field. The indexed document includes the `passage_text` field, which contains the original text, and the `passage_embedding` field, which contains the vector embeddings. 

## Step 4: Search the index using neural sparse search

To perform a neural sparse search on your index, use the `neural_sparse` query clause in [Query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/index/) queries. 

> The following example request uses a `neural_sparse` query to search for relevant documents using raw query text:

Reviewer: ```suggestion
The following example request uses a `neural_sparse` query to search for relevant documents using a raw text query:
```
